### PR TITLE
#1 initialize and configure security

### LINKED ISSUE #1: Initialize security service
**Need to:** 
- [x] Initialize SpringBoot project
- [x] Configure Spring Security
- [x] Add simple login endpoint
- [x] Configure JWT login

### PR DESCRIPTION
**Pull request by rostyslavdiachuk**
Close #1 Initialize and configure security
 
- Initialize SpringBoot project
- Configure Spring Security
- Add simple login endpoint
- Configure JWT login
- Add IDEA config for formatting
- Add TODO for future

Signed-off-by: Rdiachuk92 <rdiachuk92@gmail.com>

### CODE REVIEW

File: src/main/java/inc/sebek/securityservice/configuration/SecurityConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import org.springframework.security.web.server.SecurityWebFilterChain;
+import reactor.core.publisher.Mono;
+
+@EnableWebFluxSecurity
+@EnableReactiveMethodSecurity
+public class SecurityConfiguration {
+    private AuthenticationManager authenticationManager;
+    private SecurityContextRepository securityContextRepository;
+
+    public SecurityConfiguration(final AuthenticationManager authenticationManager,
```
Comment (cuteBrick) [inline src/main/java/inc/sebek/securityservice/configuration/SecurityConfiguration.java:20]: Isn`t a @Autowired on constructor needed?

File: src/main/java/inc/sebek/securityservice/security/AuthenticationManager.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            var roles = jwtUtil.getAllClaimsFromToken(authToken)
+                               .get("role", getListOfStringsClass())
+                               .stream()
+                               .map(Roles::valueOf)
+                               .collect(toList());
+            var auth = new UsernamePasswordAuthenticationToken(username, null, roles.stream()
+                                                                                    .map(Roles::mapToGrantedAuthority)
+                                                                                    .collect(toList()));
+            return Mono.just(auth);
+        } else {
```
Comment (cuteBrick) [inline src/main/java/inc/sebek/securityservice/security/AuthenticationManager.java]: If you put body of else statement first you will not need else statement and code will be cleaner.

File: src/main/java/inc/sebek/securityservice/security/PBKDF2Encoder.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    @Value("${spring.password.encoder.iteration}")
+    private Integer iteration;
+
+    @Value("${spring.password.encoder.keylength}")
+    private Integer keylength;
+
+    @SneakyThrows
+    @Override
+    public String encode(CharSequence rawPassword) {
+        var encodedPassword = SecretKeyFactory.getInstance("PBKDF2WithHmacSHA512")
```
Comment (cuteBrick) [inline src/main/java/inc/sebek/securityservice/security/PBKDF2Encoder.java]: Magic String

File: src/main/java/inc/sebek/securityservice/service/impls/AuthServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+    public Mono<String> getTokenForUser(String username, String password) {
+        return userService.findByUsername(username)
+                          .filter(user -> isPasswordCorrect(password, user.getPassword()))
+                          .map(user -> jwtUtil.generateToken(user));
+
+    }
+
+    private boolean isPasswordCorrect(String rawPassword, String encodedPassword) {
+        return passwordEncoder.encode(rawPassword).equals(encodedPassword);
```
Comment (cuteBrick) [inline src/main/java/inc/sebek/securityservice/service/impls/AuthServiceImpl.java]: Doesn`t passwordEncoder have dedicated method for this?

File: src/main/java/inc/sebek/securityservice/configuration/SecurityConfiguration.java (hunk 1 above)
Comment (rostyslavdiachuk) [reply src/main/java/inc/sebek/securityservice/configuration/SecurityConfiguration.java:20]: No, it is not needed. When you have only one constructor in Component(Service,Controller, etc.), Spring automaticaly know that it has `@Autowired`.

File: src/main/java/inc/sebek/securityservice/service/impls/AuthServiceImpl.java (hunk 4 above)
Comment (rostyslavdiachuk) [reply src/main/java/inc/sebek/securityservice/service/impls/AuthServiceImpl.java]: Good point

File: src/main/java/inc/sebek/securityservice/security/AuthenticationManager.java (hunk 2 above)
Comment (rostyslavdiachuk) [reply src/main/java/inc/sebek/securityservice/security/AuthenticationManager.java]: Done

File: src/main/java/inc/sebek/securityservice/security/PBKDF2Encoder.java (hunk 3 above)
Comment (rostyslavdiachuk) [reply src/main/java/inc/sebek/securityservice/security/PBKDF2Encoder.java]: Done

File: src/main/java/inc/sebek/securityservice/service/impls/AuthServiceImpl.java (hunk 4 above)
Comment (rostyslavdiachuk) [reply src/main/java/inc/sebek/securityservice/service/impls/AuthServiceImpl.java]: Done

**Review by cuteBrick — APPROVED** [review]

**Pull request merged**